include/envoy/grpc/google_grpc_creds.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "envoy/api/v2/core/grpc_service.pb.h"
+#include "envoy/common/pure.h"
+
+namespace Envoy {
+namespace Grpc {
+
+/**
+ * Interface for all Google gRPC credentials factories.
+ */
+class GoogleGrpcCredentialsFactory {
+public:
+  virtual ~GoogleGrpcCredentialsFactory() {}
+
+  /**
+   * Get a ChannelCredentials to be used for authentication of a gRPC channel.
+   * @param grpc_service_config contains configuration options
+   * @return std::shared_ptr<grpc::ChannelCredentials> to be used to authenticate a Google gRPC
+   * channel.
+   */
+  virtual std::shared_ptr<grpc::ChannelCredentials>
```

Comment:
I agree with @dio on generalizing to all creds here.

Reply:
I've added another credential factory implementation that shows how to compose multiple credential types.